stream/src/main/scala/org/apache/pekko/stream/impl/fusing/MapConcat.scala

```diff
@@ -0,0 +1,90 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.pekko.stream.impl.fusing
+
+import scala.util.control.Exception.Catcher
+import scala.util.control.NonFatal
+
+import org.apache.pekko
+import pekko.annotation.InternalApi
+import pekko.stream.{ Attributes, FlowShape, Inlet, Outlet, Supervision }
+import pekko.stream.ActorAttributes.SupervisionStrategy
+import pekko.stream.Attributes.SourceLocation
+import pekko.stream.Supervision.Decider
+import pekko.stream.impl.Stages.DefaultAttributes
+import pekko.stream.stage.{ GraphStage, GraphStageLogic, InHandler, OutHandler }
+import pekko.util.ccompat._
+
+/**
+ * INTERNAL API
+ */
+@InternalApi
+@ccompatUsedUntil213
+private[pekko] final class MapConcat[In, Out](f: In => IterableOnce[Out])
+    extends GraphStage[FlowShape[In, Out]] {
+  require(f != null, "f function should not be null")
```

Comment:
It seems to be defensive, is it nessnary check?
We should specify the parent of the function, like `f function of MapConcat should not be null` ?

Reply:
Yes，better than a materilixation error.
As the stack trace can show the detail, that's information can be ommited